File: Features/schedulablestate-heartbeat/contracts/src/test/java/net/corda/samples/heartbeat/states/StateTests.java
package net.corda.samples.heartbeat.states;

import net.corda.core.contracts.ContractState;
import net.corda.core.contracts.SchedulableState;
import net.corda.core.identity.CordaX500Name;
import net.corda.testing.core.TestIdentity;
import org.junit.Test;

import static org.jgroups.util.Util.assertFalse;
import static org.jgroups.util.Util.assertTrue;

public class StateTests {

    private final TestIdentity a = new TestIdentity(new CordaX500Name("Alice", "", "GB"));
    private final TestIdentity b = new TestIdentity(new CordaX500Name("Bob", "", "GB"));

    @Test
    public void constructorTest() {
        HeartState st = new HeartState(a.getParty());

        assertTrue(st.getParticipants().contains(a.getParty()));
        assertFalse(st.getParticipants().contains(b.getParty()));
    }

    @Test
    public void stateImplementsContractStateTest() {
        HeartState st = new HeartState(a.getParty());
        assertTrue(st instanceof ContractState);
        assertTrue(st instanceof SchedulableState);
    }
}


//public class StateTests {
//    private final MockServices ledgerServices = new MockServices();
//
//    @Test
//    public void hasFieldOfCorrectType() throws NoSuchFieldException {
////        // Does the message field exist?
////        HeartState.class.getDeclaredField("me");
////        // Is the message field of the correct type?
////        assert(HeartState.class.getDeclaredField("me").getType().equals(Party.class));
//    }
//}

File: kotlin-samples/Accounts/sharestatewithaccount/contracts/src/test/java/net/corda/samples/supplychain/contracts/StateTests.kt
package net.corda.samples.supplychain.contracts

import groovy.util.GroovyTestCase.assertEquals
import net.corda.samples.supplychain.states.InvoiceState
import net.corda.testing.node.MockServices
import org.junit.Test

class StateTests {
    private val ledgerServices = MockServices()

    @Test
    fun hasMessageFieldOfCorrectType() {
        // Does the message field exist?
        InvoiceState::class.java.getDeclaredField("amount")
        // Is the message field of the correct type?
        assertEquals(InvoiceState::class.java.getDeclaredField("amount").type, Int::class.java)
    }
}